### ISSUE TITLE
Streaming live video from sites using JW Player

### ISSUE THREAD

**Issue by IPreferGitlab** [ytdl-org/youtube-dl#22857]
## Checklist

- [x] I'm reporting a site feature request
- [x] I've verified that I'm running youtube-dl version **2019.10.22**
- [x] I've searched the bugtracker for similar site feature requests including closed ones

## Description

<!--
Provide an explanation of your site feature request in an arbitrary form. Please make sure the description is worded well enough to be understood, see https://github.com/ytdl-org/youtube-dl#is-the-description-of-the-issue-itself-sufficient. Provide any additional information, suggested solution and as much context and examples as possible.
--!>

Some news sites using JW Player can stream video well, but the live video streams, also using JW Player, cannot be streamed. This issue is quite clear on Euronews where a video such as https://www.euronews.com/2019/10/25/the-brexit-saga-continues-in-this-week-s-state-of-the-union can be streamed just fine, but the live feed(available at https://www.euronews.com/live) does not. The video tag for JW Player on the live rather than having a video url (as normal) or a playlist url, has a blob. A stream playlist can be found by using the browser network monitor (atleast under firefox), where the playlist.m3u8 url can be streamed. However, this does not suit my needs as the playlist url changes occasionally and constantly updating it is impractical. A typical playlist.m3u8 system occurs on other non-JW Player sites as well, such as https://www.vesti.ru/onair.

**Comment by uno20001** [comment]
I managed to watch the livestream in VLC using `youtube-dl --hls-prefer-native -o - https://euronews-en-p7-cdn.hexaglobe.net/f4aeaa6a7cf35a789e6631e8d52144bf/5dbcc4ef/euronews/euronews-euronews-website-web-responsive-2/ewnsabrenpri_eng.smil/playlist.m3u8 | vlc -`. Is this what you want?

**Comment by IPreferGitlab** [comment]
This would, but the issue is, the url for the playlist.m3u8 file for a stream changes on a frequent basis with JW Player, and youtube-dl is unable to locate where the url is changed to. Thus after a period of time, it simply replies with a 403 error.

**Comment by uno20001** [comment]
The normal videos work because the page has an `og:video` meta tag, which contains the url of the actual video file, so the generic extractor can pick it out. The livestream page is quite different, the page queries (using AJAX) [a JSON file](https://euronews.com/api/watchlive.json), which contains an url to another JSON file which at last contains the url to the playlist. I don't think that it is reasonable to expect `youtube_dl` to extract the actual playlist url behind so many layers. I'm also not sure if completely new extractor is justified only to be able to handle euronews's livestream.

**Comment by IPreferGitlab** [comment]
This occurs on any JW Player livestream, which in my experience is used in the majority of news sites. and other broadcasters. See also, Comet TV and Charge TV as examples of US broadcasters who use it(I believe this is what MGM TV uses as a standard, but I may be mistaken).

**Comment by uno20001** [comment]
Comet TV works for me with `youtube_dl -v -o - --hls-prefer-native https://www.comettv.com/watch-live | vlc -` (I get the "This stream is not available in your are" both in the browser and with youtube_dl, but it works nonetheless). Charge TV works as well.

**Comment by IPreferGitlab** [comment]
Yes. Stream errors such as not available in region work. But the actual stream does not load. You can use a VPN and see for yourself. It fails like any other JW Player livestream.
